### PR TITLE
String_200

### PR DESCRIPTION
**Pull request by madhan0923**
This file is used to split the variables in a dataframe which has length > 200 into meaningful string variable with SUPP variables

Thank you for your Pull Request! We have developed this task checklist from the
[Development Process
Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html)
to help with the final steps of the process. Completing the below tasks helps to
ensure our reviewers can maximize their time on your code as well as making sure
the oak codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/split_var_call.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
+R_SPLIT = function(domain_dataset,max_length_out = 200){
```
Comment (galachad) [inline R/split_var_call.R:1]: - Please use tidyver styleguid and snake case https://style.tidyverse.org/syntax.html
- Please add roxygen documentation, with examples
- Please add tests cases
- Please add namesacpes:: to all functions calls from tidyverse packages

File: R/split_var_call.R [hunk 2]
```diff
@@ -0,0 +1,60 @@
+R_SPLIT = function(domain_dataset,max_length_out = 200){
+  out_n = outt = NULL
+
+  #filtering columns > 200
+  char_200 = domain_dataset %>% select_if(~ max(nchar(.)) >= max_length_out)
+
+  #string split function
+  split_var <- function(string,max_length_out = 200) {
```
Comment (galachad) [inline R/split_var_call.R:8]: Please do not keep function inside function.

File: R/split_var_call.R [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  out_n = outt = NULL
+
+  #filtering columns > 200
+  char_200 = domain_dataset %>% select_if(~ max(nchar(.)) >= max_length_out)
+
+  #string split function
+  split_var <- function(string,max_length_out = 200) {
+
+    # Pattern spot
+    pattern = names(which.max(table(str_extract_all(string, "[:punct:]|[:blank:]")))) %>%
```
Comment (galachad) [inline R/split_var_call.R:11]: The pattern can be used for split, but it will not work as sep in paste function.

File: R/split_var_call.R [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  #FUNCTION CALL
+  outt <- map(char_200, ~ {
+    split_list <- map(.x, ~ {
+      cv <- as.data.frame(split_var(.x, max_length_out))
+      names(cv) <- seq_along(cv)
+      cv
+    })
+    split_df <- bind_rows(split_list)
+    split_df
+  }) %>% imap(.,~set_names(.x,.y)) %>% bind_cols()
```
Comment (galachad) [inline R/split_var_call.R:54]: Can we consider use of tidyr?

**Review by galachad — CHANGES_REQUESTED** [review]
PTAL, I added some of initial comments. @madhan0923 can we consider of using S3 methods for diffrent classes?

**Comment by rammprasad** [comment]
@madhan0923 - Can you address the review comments? Also, merge main to this FB and make sure the pipeline is green.
